src/main/java/com/teragrep/cfe_39/consumers/kafka/DatabaseOutput.java

```diff
@@ -0,0 +1,394 @@
+        this.eventNodeSourceSource = new SDVector("event_node_source@48577","source");
+        this.eventNodeRelaySource = new SDVector("event_node_relay@48577","source");
+        this.eventNodeSourceSourceModule = new SDVector("event_node_source@48577","source_module");
+        this.eventNodeRelaySourceModule = new SDVector("event_node_relay@48577","source_module");
+        this.eventNodeSourceHostname = new SDVector("event_node_source@48577","hostname");
+        this.eventNodeRelayHostname = new SDVector("event_node_relay@48577","hostname");
+        this.originHostname = new SDVector("origin@48577","hostname");
+//        this.approximatedSize = 0;
+        this.epochMicros_last = 0L;
+    }
+
+    // Checks that the filesize stays under the defined maximum file size. If the file is about to go over target limit commits the file to HDFS and returns true, otherwise does nothing and returns false.
+    boolean committedToHdfs(long fileSize, RecordOffset recordOffsetObject) {
+        try {
+            // If the syslogAvroWriter is already initialized, check the filesize so it doesn't go above maximumFileSize.
+            if (fileSize > maximumFileSize) {
+                // file too large for adding the new record to it, write the still adequately sized AVRO-file to the HDFS database and create a new empty AVRO-file.
+
+                // This part closes the writing of now "complete" AVRO-file and stores the file to HDFS.
+                syslogAvroWriter.close();
+                try (HDFSWrite writer = new HDFSWrite(config, recordOffsetObject)) {
+                    writer.commit(syslogFile, epochMicros_last); // commits the final AVRO-file to HDFS.
+                }
+
+                // This part defines a new empty file to which the new AVRO-serialized records are stored until it again hits the 64M size limit.
+                writableQueue.setQueueNamePrefix(recordOffsetObject.getTopic()+recordOffsetObject.getPartition());
+                syslogFile = writableQueue.getNextWritableFile();
+                syslogAvroWriter = new SyslogAvroWriter(syslogFile);
+//                approximatedSize = syslogAvroWriter.getFileSize(); // resets the size approximation.
+                return true;
+            }
+        } catch (IOException ioException) {
+            throw new UncheckedIOException(ioException);
+        }
+        return false;
+    }
+
+    private long rfc3339ToEpoch(ZonedDateTime zonedDateTime) {
+        final Instant instant = zonedDateTime.toInstant();
+
+        final long MICROS_PER_SECOND = 1000L * 1000L;
+        final long NANOS_PER_MICROS = 1000L;
+        final long sec = Math.multiplyExact(instant.getEpochSecond(), MICROS_PER_SECOND);
+
+        return Math.addExact(sec, instant.getNano() / NANOS_PER_MICROS);
+    }
+
+    // Input parameter is a list of RecordOffsetObjects. Each object contains a record and its metadata (topic, partition and offset).
```

Comment:
please use block comments for continued sections

Reply:
Solved in commit 022832c8b423efc7466b7aead56cc317a45a0002